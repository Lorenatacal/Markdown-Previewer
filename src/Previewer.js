import React from 'react';
import ReactDOM from 'react-dom';
import Text from './Text'
import styled from 'styled-components';
import ReactMarkdown from 'react-markdown';


const StyledContainer = styled.div`
float:right;
align:left;
width: 45%;
height: 400px;
padding: 12px 20px;
box-sizing: border-box;
border: 2px solid #ccc;
border-radius: 4px;
background-color: #f8f8f8;
resize: none;
border-radius: 20px;
padding: 0.25em;
margin: 6em;
margin-left: 25px;
margin-right: 25px;
align: center;
text-align: center;
font-family: arial;
`;

const StyledTitle = styled.h3`
text-align: center;
`;

const StyledReactMarkdown = styled(ReactMarkdown)`
width: 98%;
height: 70%;
overflow-wrap: break-word;
`;

const Previewer = props => {
    return (
        <StyledContainer>
            <StyledTitle>Previewer</StyledTitle>
            <StyledReactMarkdown source={props.input} />
        </StyledContainer>
    )
}

export default Previewer;