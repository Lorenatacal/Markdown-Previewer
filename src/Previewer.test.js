import React, {Component} from 'react';
import Previewer from './Previewer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import { ENETRESET } from 'constants';
 
Enzyme.configure({ adapter: new Adapter() });

test("Previewer should render correctly with data", () => {
    const value = {
        value: '# Welcome to the Markdown Previewer\n\n## The title was a header, this is a sub-header\n\n### **Ussage:**\n\n * Write markdown text in this textarea.\n * See the changes in the previewer\n\n### **Text attributes in number list:**\n\n1. *italic*,\n2. **bold**,\n3. **_italic-bold_**\n4. `monospace`\n5. ~~strikethrough~~ \n\n*[Checkout my gitHub for more projects](https://github.com/Lorenatacal)*'
    }
    const wrapper = Enzyme.shallow(<Previewer source={value} />)
    expect(toJson(wrapper)).toMatchSnapshot();
})