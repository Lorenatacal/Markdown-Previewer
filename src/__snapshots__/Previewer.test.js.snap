// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previewer should render correctly with data 1`] = `
<Previewer
  value={
    Object {
      "value": "# Welcome to the Markdown Previewer
    
    ## The title was a header, this is a sub-header
    
    ### **Ussage:**
    
     * Write markdown text in this textarea.
     * See the changes in the previewer
    
    ### **Text attributes in number list:**
    
    1. *italic*,
    2. **bold**,
    3. **_italic-bold_**
    4. \`monospace\`
    5. ~~strikethrough~~ 
    
    *[Checkout my gitHub for more projects](https://github.com/Lorenatacal)*",
    }
  }
>
  <styled.div>
    <div
      className="sc-bxivhb juhBrH"
    >
      <Styled(ReactMarkdown)>
        <ReactMarkdown
          astPlugins={Array []}
          className="sc-ifAKCX cbJEgc"
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          sourcePos={false}
          transformLinkUri={[Function]}
        >
          <Root
            className="sc-ifAKCX cbJEgc"
            key="root-1-1"
          >
            <div
              className="sc-ifAKCX cbJEgc"
            />
          </Root>
        </ReactMarkdown>
      </Styled(ReactMarkdown)>
    </div>
  </styled.div>
</Previewer>
`;
